core/src/main/scala/com/pingcap/tispark/auth/TiAuthorization.scala

```diff
@@ -0,0 +1,282 @@
+package com.pingcap.tispark.auth
+
+import com.pingcap.tikv.{TiConfiguration, TiDBJDBCClient}
+import com.pingcap.tispark.TiDBUtils
+import com.pingcap.tispark.auth.TiAuthorization.{logger, parsePrivilegeFromRow, refreshInterval}
+import com.pingcap.tispark.write.TiDBOptions
+import org.apache.spark.sql.internal.SQLConf
+import org.slf4j.LoggerFactory
+
+import java.sql.SQLException
+import java.util.concurrent.atomic.AtomicReference
+import java.util.concurrent.locks.ReentrantLock
+import java.util.concurrent.{Executors, ScheduledExecutorService, TimeUnit}
+import scala.collection.JavaConverters
+import scala.util.control.Breaks.{break, breakable}
+
+case class TiAuthorization private (parameters: Map[String, String], tiConf: TiConfiguration) {
+
+  private var tiDBJDBCClient: TiDBJDBCClient = _
+
+  private val scheduler: ScheduledExecutorService =
+    Executors.newScheduledThreadPool(1)
+
+  val globalPriv: AtomicReference[List[MySQLPriv.Value]] =
+    new AtomicReference(List())
+
+  val databasePrivs: AtomicReference[Map[String, List[MySQLPriv.Value]]] =
+    new AtomicReference(Map())
+
+  val tablePrivs: AtomicReference[Map[String, Map[String, List[MySQLPriv.Value]]]] =
+    new AtomicReference(Map())
+
+  val task: Runnable = () => {
+    val privs = getPrivileges
+    globalPriv.getAndSet(privs.globalPriv)
+    databasePrivs.getAndSet(privs.databasePrivs)
+    tablePrivs.getAndSet(privs.tablePrivs)
+  }
+
+  {
+    TiAuthorization.dbPrefix = tiConf.getDBPrefix
+    val option = new TiDBOptions(parameters)
+    try {
+      this.tiDBJDBCClient = new TiDBJDBCClient(TiDBUtils.createConnectionFactory(option.url)())
+    } catch {
+      case e: Throwable => {
+        logger.error(f"Failed to create tidb jdbc client with url ${option.url}", e)
+        System.exit(-1)
```

Comment:
I'm not sure, but this looks strange.

Reply:
In my opinion, if the authentication fail, the connection should shutdown

Reply:
> In my opinion, if the authentication fails, the connection should shut down, what's your opinion, throw an exception?

throw an exception is a better option

Reply:
Sorry that I didn't make it clear that shutdown JVM looks strange.